airbyte-integrations/connectors/source-notion/source_notion/manifest.yaml

```diff
@@ -0,0 +1,399 @@
+        class_name: source_notion.components.NotionPropertiesTransformation
+
+  - type: DeclarativeStream
+    name: pages
+    primary_key:
+      - id
+    retriever:
+      type: SimpleRetriever
+      requester:
+        type: HttpRequester
+        url_base: https://api.notion.com/v1/
+        path: search
+        http_method: POST
+        request_parameters: {}
+        request_headers:
+          Notion-Version: "2022-06-28"
+        authenticator:
+          authenticator:
+          type: BearerAuthenticator
+          api_token: "{{ config.get('credentials', {}).get('token') if config.get('credentials', {}).get('auth_type') == 'token' else config.get('credentials', {}).get('access_token') if config.get('credentials', {}).get('auth_type') == 'OAuth2.0' else config.get('access_token', '') }}"
+        error_handler:
+          type: CompositeErrorHandler
+          error_handlers:
+            - type: DefaultErrorHandler
+              backoff_strategies:
+                - type: WaitTimeFromHeader
+                  header: retry-after
+              response_filters:
+                - type: HttpResponseFilter
+                  action: RETRY
+                  http_codes:
+                    - 429
+            - type: DefaultErrorHandler
+              response_filters:
+                - type: HttpResponseFilter
+                  action: IGNORE
+                  http_codes:
+                    - 400
+                  error_message_contains: "The start_cursor provided is invalid:"
+        request_body_json:
+          sort: '{ "direction": "descending", "timestamp": "last_edited_time" }'
+          filter: '{ "property": "object", "value": "page" }'
+      record_selector:
+        type: RecordSelector
+        extractor:
+          type: DpathExtractor
+          field_path:
+            - results
+      paginator:
+        type: DefaultPaginator
+        page_token_option:
+          type: RequestOption
+          inject_into: body_json
+          field_name: start_cursor
+        page_size_option:
+          type: RequestOption
+          field_name: page_size
+          inject_into: body_json
+        pagination_strategy:
+          type: CursorPagination
+          page_size: 100
+          cursor_value: '{{ response.get("next_cursor", {}) }}'
+          stop_condition: '{{ not response.get("next_cursor", {}) }}'
+      partition_router: []
+    incremental_sync:
+      type: DatetimeBasedCursor
+      cursor_field: last_edited_time
+      cursor_datetime_formats:
+        - "%Y-%m-%dT%H:%M:%S.%fZ"
+      datetime_format: "%Y-%m-%dT%H:%M:%S.%fZ"
+      start_datetime:
+        type: MinMaxDatetime
+        datetime: "{{ config['start_date'] }}"
+        datetime_format: "%Y-%m-%dT%H:%M:%S.%fZ"
+      is_data_feed: true
+    transformations:
+      - type: CustomTransformation
+        class_name: source_notion.components.NotionPropertiesTransformation
+
+  - type: DeclarativeStream
+    name: comments
```

Comment:
Should this be using the `NotionPropertiesTransformation`? Looks like it was [previously using ](https://github.com/airbytehq/airbyte/pull/35974/files#diff-c9b27094ad0140b4f070897905b197dbe40328e56792e12b8e02893e622e5890L436)`transform_properties`.

Reply:
This one's a bit of an oddity that is entirely my bad. Comments records don't have a `properties` key, so the method was only ever returning the records without any alterations. This was actually the first stream I implemented when I signed on last year (hence the questionable choices 🫠), and if memory serves I was encountering failures at the time when trying to simply yield the record that were resolved when I added the method. I can only assume I was making some unrelated noob syntax mistakes and settled for the first solution that got things working again, as this method is indeed superfluous and everything works exactly the same when it is removed.

Reply:
All good! Thanks for the clarification.